kotlin/src/main/kotlin/io/smallrye/mutiny/coroutines/Multi.kt

```diff
@@ -50,36 +41,25 @@ suspend fun <T> Multi<T>.asFlow(): Flow<T> = channelFlow<T> {
 
         override fun onItem(item: T) {
             if (parentCtx.isActive) {
-                runBlocking {
-                    channel.send(item)
-                }
-            } else if (terminationLock.isLocked) {
+                channel.trySendBlocking(item).getOrThrow()
+            } else {
                 // Coroutine is completed or was cancelled:
                 // regular cancellation may happen by Flow abortion like with Flow.first(), no failure must be thrown
-                // in case of a failure, 'invokeOnClose' is called with a reason separately - setting a terminatinoFailure is also not necessary here
                 subscription.get()?.cancel()
-                terminationLock.unlock()
             }
         }
 
         override fun onFailure(failure: Throwable) {
-            terminationFailure.set(failure)
-            terminationLock.unlock()
+            cancel(CancellationException(failure.message, failure))
         }
 
         override fun onCompletion() {
-            if (terminationLock.isLocked) {
-                terminationLock.unlock()
-            }
+            channel.close()
         }
-
     }
-
     subscribe().withSubscriber(subscriber)
-    // wait (suspending) for completion or failure of Multi to terminate the Flow
-    terminationLock.lock()
-    terminationFailure.get()?.also { throw it }
-}
+    awaitClose()
+}.buffer(Channel.UNLIMITED) // Switching to unlimited channel capacity avoids blocking
```

Comment:
How does that work with back-pressure? There's a OOM exception risk with unlimited buffering.

Reply:
like that idea, thank you.
shall we do this only for Mutiny 2, as it comes with a method signature change, that needs a re-compliation of the using sources?

(kotlin default parameters are source compatible, but not bytecode compatible)

Reply:
We can afford to introduce a minor compatibility breakage it in 1.x for the Kotlin support as long as it is documented.

Another option is to keep the current method and document that it will perform unlimited buffering, and have another name for the other method(s), e.g., `.toFlowUsing(config)`, or `.toFlowAndBuffer(256)` (again must an idea, I'm sure you'll have a better idea 🤣 )

Reply:
it's hard to assess the quality of an idea at the idea state ;)

how about simply make the buffer parameters available for `toFlow` using sensible defaults - updated the PR with that.

Reply:
Ah yes that works quite nice with default parameters 😄